SupportYourLocals.WPF/MainWindow.xaml.cs

```diff
@@ -653,6 +656,237 @@ private void DisplayErrorMessage(string message)
+
+                GridLogin.Visibility = Visibility.Collapsed;
+                GridRegistration.Visibility = Visibility.Collapsed;
+            }
+
+            GridUserData.Visibility = Visibility.Collapsed;
+            GridUserDataToLogin.Visibility = Visibility.Collapsed;
+        }
+
+        private void LoginWhenRegistered_Clicked(object sender, RoutedEventArgs e)
+        {
+            GridLogin.Visibility = Visibility.Visible;
+            GridRegistration.Visibility = Visibility.Collapsed;
+        }
+
+        public bool Duplicates(int randomID)
+        {
+            foreach (var userID in userLoginData.GetAllData())
+            {
+                if (randomID.Equals(userID.ID))
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public bool CheckRegisterData(string password, string username)
+        {
+            var users = userLoginData.GetAllData();
+            var usernameExists = false;
+
+            foreach (var name in users)
+            {
+                usernameExists = (name.Username == username);
+            }
+
+            if (password.Length == 0 || username.Length == 0 || ConfirmPasswordBoxR.Password.Length == 0)
+            {
+                LabelForEmptyFieldsError.Visibility = Visibility.Visible;
+                return false;
+            }
+            else if (usernameExists)
+            {
+                LabelForUsedUsernameError.Visibility = Visibility.Visible;
+            }
+            else
+            {
+                if(!CheckUsernameRegex(username))
+                {
+                    LabelForInvalidUsernameError.Visibility = Visibility.Visible;
+                }
+                else // if username is correct
+                {
+                    if(!CheckPasswordLength(password))
+                    {
+                        LabelForPasswordError.Visibility = Visibility.Visible;
+                    }
+                    else if(CheckConfirmPassword(PasswordBoxR.Password, ConfirmPasswordBoxR.Password)) // if password is correct
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        private bool CheckPasswordLength(String password) => password.Length >= 6;
+
+        private bool CheckUsernameRegex(String username)
+        {
+            Regex regexUsername = new Regex(@"^[a-zA-ZĄ-ž0-9-. ]+$");
+
+            if (!regexUsername.IsMatch(username))
+            {
+                return false;
+            }
+            return true;
+        }
+
+        private bool CheckConfirmPassword(string password, string confirmPassword)
+        {
+            if (password == confirmPassword)
+            {
+                return true;
+            }
+
+            LabelForConfirmPasswordError.Visibility = Visibility.Visible;
+            return false;
+        }
```

Comment:
I don't think this needs to be a separate function, since this is just an equality comparison.